packages/salesforcedx-vscode-core/src/commands/forceSourceDiff.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  DiffResultParser,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { nls } from '../messages';
+import { notificationService, ProgressNotification } from '../notifications';
+import { taskViewService } from '../statuses';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath, hasRootWorkspace, OrgAuthInfo } from '../util';
+import {
+  FilePathGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+export class ForceSourceDiffExecutor extends SfdxCommandletExecutor<string> {
+  public build(filePath: string): Command {
+    const commandBuilder = new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_source_diff_text'))
+      .withArg('force:source:diff')
+      .withLogName('force_source_diff')
+      .withFlag('--sourcepath', filePath)
+      .withJson();
+    return commandBuilder.build();
+  }
+
+  public async execute(response: ContinueResponse<string>): Promise<void> {
```

Comment:
Can we break this down a little bit to simplify this method?

Reply:
I could move out the contents of `execution.processExitSubject.subscribe ...` but the rest I can't since we're overriding the `execute` provided by `SfdxCommandletExecutor`.